### PR TITLE
Add kingping integration

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

File: https/tls_config.go [hunk 1]
```diff
@@ -164,8 +166,23 @@ func ConfigToTLSConfig(c *TLSStruct) (*tls.Config, error) {
 	return cfg, nil
 }
 
-// Listen starts the server on the given address. If tlsConfigPath isn't empty the server connection will be started using TLS.
+// AddFlags adds the flags used by this package to the Kingpin application.
+// To use the default Kingpin application, call AddFlags(kingpin.CommandLine)
+func AddFlags(a *kingpin.Application) {
```
Comment (SuperQ) [inline https/tls_config.go:171]: Let's call this `AddKingpinFlags`, as we could include other flag library support in the future.

File: https/tls_config.go [hunk 2]
```diff
@@ -164,8 +166,23 @@ func ConfigToTLSConfig(c *TLSStruct) (*tls.Config, error) {
...
+	a.Flag(
+		"web.config",
+		"Path to config yaml file that can enable TLS or authentication.",
+	).Default("").StringVar(&cliWebConfigFlag)
+}
+
+// Listen starts the server on the given address. If tlsConfigPath or
```
Comment (SuperQ) [inline https/tls_config.go:178]: > If tlsConfigPath or tlsConfigPath aren't empty

Wat?

**Comment by roidelapluie** [comment]
@SuperQ WDYT?

File: https/tls_config.go [hunk 3]
```diff
@@ -164,14 +162,21 @@ func ConfigToTLSConfig(c *TLSStruct) (*tls.Config, error) {
 	return cfg, nil
 }
 
-// Listen starts the server on the given address. If tlsConfigPath isn't empty the server connection will be started using TLS.
-func Listen(server *http.Server, tlsConfigPath string, logger log.Logger) error {
-	if tlsConfigPath == "" {
+// TLSConfig contains the configuration required to start an tls/basic auth server.
```
Comment (SuperQ) [inline https/tls_config.go:165]: I'd like this to be more clear that it's a path to a yaml file.

Do we need to get complex enough that this is a struct?

Maybe just something like this:

```
type TLSConfigFilePath string
```

**Comment by roidelapluie** [comment]
@SuperQ WDYT now?

**Review by SuperQ — APPROVED** [review]
Simple and easy, I like it.

**Pull request merged**